crates/forge/bin/main.rs

```diff
@@ -103,3 +106,25 @@ fn main() -> Result<()> {
         },
     }
 }
+
+/// Set the program execution context based on `forge` subcommand used.
+/// The execution context can be set only once per program, and it can be checked by using
+/// cheatcodes.
+fn init_execution_context(subcommand: &ForgeSubcommand) {
+    let context = match subcommand {
+        ForgeSubcommand::Test(_) => ForgeContext::Test,
+        ForgeSubcommand::Coverage(_) => ForgeContext::Coverage,
+        ForgeSubcommand::Snapshot(_) => ForgeContext::Snapshot,
+        ForgeSubcommand::Script(cmd) => {
+            if cmd.broadcast {
+                ForgeContext::ScriptBroadcast
+            } else if cmd.resume {
+                ForgeContext::ScriptResume
+            } else {
+                ForgeContext::ScriptDryRun
+            }
+        }
+        _ => ForgeContext::Unknown,
```

Comment:
Should we exit with an error here and instruct the user to open an issue in this case? Ideally we should never hit this branch since all contexts should be enumerated

Reply:
I don't think we should, since it is matched on forge subcommand then all the other subcomands like doc, debug, etc. should be handled then